lib/index.js

```diff
@@ -243,7 +243,10 @@ internals.Client = class {
                 redirectOptions.timeout = (redirectOptions.timeout - elapsed).toString();           // stringify to not drop timeout when === 0
             }
 
-            const followRedirect = () => {
+            const followRedirect = (err) => {
+                if (err) {
+                    return finishOnce(Boom.badGateway('Invalid redirect: ' + err.message, _trace));
```

Comment:
We could preserve the user's error and also treat it as a bad gateway using `Boom.boomify()`.  It also handles the error message prefixing 👍